<h1 style="color:black"> Variables </h1>

   Es un espacio reservado de memoria que tiene la capacidad de cambiar de valor mientras se desarrolla la ejecución del programa y se le asigna un <b> nombre </b>.

101 True Hola! 1.1 (1.1+3.2j)


<class 'int'>
<class 'bool'>
<class 'str'>
<class 'float'>
<class 'complex'>


<h1 style="color:black"> Operadores aritméticos </h1>


   | Caracter | Operación |
   |----|---|
   | +  | Suma |
   | -  | Resta |
    | /  | division |
    | %  | módulo |
    | *  | multiplicación |
    | //  | Divisón floor |
    | **  | Elevado a la potencia |
   

True

False

8 2 4


<h1 style="color:black"> Operadores Lógicos </h1>

   | Caracter | Operación |
   |----|---|
   | & | AND |
   | l  | OR |
    | ^  | XOR |
    | ~ | Negación |


False
True


<h1 style="color:black"> Range </h1>

Crea una lista de <b> enteros </b> entre cierto rango definido por un intervalo

range(0, 3)
range(2, 9)
range(2, 27, 8)


range

<h1 style="color:black"> Print </h1>

```print()``` Imprime por pantalla diferentes textos

In [66]:
#Concatenación


Hola Mundo! en una linea
Hola 
Mundo!
El valor de la variable es 123
El valor de pi es 3.14


<h1 style="color:black"> Estructura de Datos Básicas en Python </h1>

Es una forma de ordenar y agrupar particulares estructuras de información

## Listas

Listas son uno de las estructuras de datos más usadas. Es una secuencia de información ordenada que se encuentra entre brackets y separado por una coma. Cada uno de estos elementos en las listas son accesibles por un índice.

Para crear una lista vacia , solo es necesario definir una variable y inicializar 

```mi_lista = []```

In [68]:
mi_lista = []
type(mi_lista)

list

Es posible asignar de forma directa una lista

['naranja', 'papa', 12.42]


### Indexing

Los índices en Python comienzan en cero. "mi_lista1" tiene 3 elements "0" -> "naranja", "1" -> "papa" , "2" -> 12.42

'naranja'

El índice puede ser ordenado dde forma inversa. "-3" -> "naranja", "-2" -> "papa" , "-1" -> 12.42

'naranja'

Podemos tambien hacer lista en listas, a continuación se ilustra

[['naranja', 'papa', 12.42], [False, (1+2j), 'Hola']]


Acceder a un elemento de una lista dentro de otra lista

'naranja'

### Slicing

Al utiliar los indíces es posible acceder a solo un valor, en cambio utilizando slicing es posible obtener una lista de una lista utilizando la siguiente notación : [a,b] donde a y b son enteros y definen un intervalo de valores

[0, 1, 2, 3]
[4, 5, 6, 7, 8, 9]


[0, 3, 6]

### Funciones de las listas

Para encontrar la longitud de la lista o el número de elementos de una lista, se utiliza len( ).

10

Si la lista está formada por todos los elementos enteros, entonces min( ) y max( ) dan el valor mínimo y máximo de la lista.

0

9

Las listas se pueden concatenar añadiendo, '+' a las mismas. La lista resultante contendrá todos los elementos de las listas que se añadieron. La lista resultante no será una lista anidada.

[1, 2, 3, 5, 4, 7]

Puede surgir un requisito en el que necesite comprobar si un elemento concreto está en una lista predefinida. Considere la siguiente lista.

Para comprobar si 'Fire' y 'Rajath' están presentes en los nombres de la lista. Un enfoque convencional sería utilizar un bucle for e iterar sobre la lista y utilizar la condición if. Pero en python puedes usar el concepto 'a in b' que devolvería 'True' si a está presente en b y 'False' si no.

True

False

### append
append( ) se utiliza para añadir un elemento al final de la lista.

[1, 1, 4, 8, 7, 1]


La función append( ) también puede utilizarse para añadir una lista completa al final. Observe que la lista resultante se convierte en una lista anidada.

[1, 1, 4, 8, 7, 1, [5, 4, 2, 8]]


### extent
Pero si la lista anidada no es lo que se desea, se puede utilizar la función **extend( )**.

[1, 1, 4, 8, 7, 1, [5, 4, 2, 8], 5, 4, 2, 8]


### index()
index( ) se utiliza para encontrar el valor del índice de un elemento concreto. Tenga en cuenta que si hay varios elementos con el mismo valor, se devuelve el primer valor de índice de ese elemento.

0

### insert(x,y)

index( ) se utiliza para encontrar el valor del índice de un elemento concreto. Tenga en cuenta que si hay varios elementos con el mismo valor, se devuelve el primer valor de índice de ese elemento.

[1, 1, 4, 8, 7, 'name', 1, [5, 4, 2, 8], 5, 4, 2, 8]


insert(x,y) inserta pero no reemplaza el elemento. Si quieres reemplazar el elemento con otro elemento simplemente asigna el valor a ese índice en particular

[1, 1, 4, 8, 7, 'Python', 1, [5, 4, 2, 8], 5, 4, 2, 8]


### pop

pop( ) se utiliza para eliminar un elemento basándose en su valor de índice, que puede asignarse a una variable. También se puede eliminar un elemento especificando el propio elemento mediante la función remove( ).

8

Se puede especificar el valor del índice para que aparezca un elemento ceratin correspondiente a ese valor del índice.

1

### Remove

pop( ) se utiliza para eliminar un elemento basándose en su valor de índice, que puede asignarse a una variable. También se puede eliminar un elemento especificando el propio elemento mediante la función remove( ).

[1, 4, 8, 7, 1, [5, 4, 2, 8], 5, 4, 2]
[1, 4, 8, 7, 1, 5, 4, 2]


### Sorting

[1, 1, 2, 4, 4, 5, 7, 8]


[8, 7, 5, 4, 4, 2, 1, 1]


### Copiando una lista

**La mayoría de los nuevos programadores de python cometen este error. Considere lo siguiente***

[2, 1, 4, 3]


Aquí, hemos declarado una lista, lista = [2,1,4,3]. Esta lista se copia a listab asignándole su valor y se copia como se ve. Ahora realizamos algunas operaciones aleatorias sobre lista.

[2, 1, 4]
[2, 1, 4, 9]


[2, 1, 4, 9]


Como Solucionarlo?

lista = [2,1,4,3]
listb = lista[:]
print(listb)
lista.pop()
print(lista)
lista.append(9)
print(lista)

[2, 1, 4, 3]


**Opción 2**

[2, 1, 4, 3]
[2, 1, 4]
[2, 1, 4, 9]
[2, 1, 4, 3]


# Tuplas

Las tuplas son similares a las listas, pero la única gran diferencia es que los elementos de una lista pueden cambiarse, pero en las tuplas no. Piensa en las tuplas como algo que tiene que ser True para algo en particular y no puede ser True para ningún otro valor. 

('hola', 'Como estan', 3.14159265)


tuple

hola Como estan 3.14159265
3.14159265


# Estructuras de Control

## if

Hello


# If-else

hello


## if-elif

x<y


x<y
x=10


# Loops

## For

0
1
2
3
4


azul
rojo
morado
etc


[1, 2, 3] 1
[1, 2, 3] 2
[1, 2, 3] 3
[4, 5, 6] 4
[4, 5, 6] 5
[4, 5, 6] 6
[7, 8, 9] 7
[7, 8, 9] 8
[7, 8, 9] 9


## while

1
4
Bye


## Break

SyntaxError: Missing parentheses in call to 'print'. Did you mean print(i)? (<ipython-input-193-96bb291e7e7e>, line 2)

## Continue

0
1
2
3
4
The end.
The end.
The end.
The end.
The end.


[27, 54, 81, 108, 135, 162, 189, 216, 243, 270]